File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/invocation/Converter.java
package org.javamaster.invocationlab.admin.service.invocation;

import org.javamaster.invocationlab.admin.service.invocation.entity.PostmanRequest;
import org.javamaster.invocationlab.admin.service.invocation.entity.RpcParamValue;
import org.javamaster.invocationlab.admin.service.invocation.exception.ParamException;

/**
 * @author yudong
 */
public interface Converter<R extends PostmanRequest, T extends RpcParamValue> {

    T convert(R request, Invocation invocation) throws ParamException;
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/invocation/Converter.kt
package org.javamaster.invocationlab.admin.service.invocation

import org.javamaster.invocationlab.admin.service.invocation.entity.PostmanRequest
import org.javamaster.invocationlab.admin.service.invocation.entity.RpcParamValue

/**
 * @author yudong
 */
interface Converter<R : PostmanRequest, T : RpcParamValue> {
    fun convert(request: R, invocation: Invocation): T
}
